components/rusoto_util/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+                secret_access_key,
+                None, /* token */
+                None, /* valid_for*/
+            ))
+        } else {
+            CredentialsProvider::Default(Box::new(
+                AutoRefreshingProvider::new(DefaultCredentialsProvider::default()).map_err(
+                    |e| {
+                        Error::new(
+                            ErrorKind::Other,
+                            format!("create aws credentials provider error: {}", e),
+                        )
+                    },
+                )?,
+            ))
+        };
+        Ok(cred_provider)
+    }
+}
+
+#[async_trait]
+impl ProvideAwsCredentials for CredentialsProvider {
+    async fn credentials(&self) -> Result<AwsCredentials, CredentialsError> {
+        match self {
+            CredentialsProvider::Default(default_provider) => default_provider.credentials().await,
+            CredentialsProvider::Static(static_provider) => static_provider.credentials().await,
+        }
+    }
+}
+
+const AWS_WEB_IDENTITY_TOKEN_FILE: &str = "AWS_WEB_IDENTITY_TOKEN_FILE";
+const AWS_ROLE_ARN: &str = "AWS_ROLE_ARN";
+const AWS_ROLE_SESSION_NAME: &str = "AWS_ROLE_SESSION_NAME";
+const DEFAULT_SESSION_NAME: &str = "WebIdentitySession";
+
+// Same as rusoto_credentials::DefaultCredentialsProvider with extra
+// rusoto_sts::WebIdentityProvider support.
+pub struct DefaultCredentialsProvider {
+    // Underlying implementation of rusoto_credentials::DefaultCredentialsProvider.
+    default_provider: ChainProvider,
+    // Provider IAM support in Kubernetes.
+    web_identity_provider: WebIdentityProvider,
+}
+
+impl Default for DefaultCredentialsProvider {
+    fn default() -> DefaultCredentialsProvider {
+        DefaultCredentialsProvider {
+            default_provider: ChainProvider::new(),
+            // We should be using WebIdentityProvider::from_k8s_env(), after the issue have been
+            // fixed: https://github.com/rusoto/rusoto/pull/1724
+            web_identity_provider: WebIdentityProvider::new(
+                // Get token file name from env var, then read from the file.
+                Variable::dynamic(|| {
+                    Variable::from_text_file(
+                        Variable::<String, CredentialsError>::from_env_var(
+                            AWS_WEB_IDENTITY_TOKEN_FILE,
+                        )
+                        .resolve()?,
+                    )
+                    .resolve()
+                }),
+                Variable::from_env_var(AWS_ROLE_ARN),
+                // As AWS_ROLE_SESSION_NAME is optional, we cannot use Variable::from_env_var.
+                Some(Variable::dynamic(|| {
+                    match var(AWS_ROLE_SESSION_NAME).map(|v| v.trim().to_owned()) {
+                        Ok(v) if !v.is_empty() => Ok(v),
+                        Ok(_) => Ok(DEFAULT_SESSION_NAME.to_owned()),
+                        Err(VarError::NotPresent) => Ok(DEFAULT_SESSION_NAME.to_owned()),
```

Comment:
```suggestion
                        Ok(_) | Err(VarError::NotPresent) => Ok(DEFAULT_SESSION_NAME.to_owned()),
```

Since `Ok("")` as a pattern also works, maybe move this line above line 143 to skip the `!v.is_empty()` check.

Reply:
`Ok("")` doesn't seems to work, as the value being matched is `Option<String>`.